### DISCUSSION TITLE
How to override delete operator in JS from C#.

### DISCUSSION THREAD

**Discussion by nathan130200** [sebastienros/jint#1426] [Q&A] [answered]
Hello,

I'm trying to create an automated script and i will expose an disposable class to JS. There's an way to proper call `delete myObj` from JS and it redirect to `myObj.Dispose()` in C#? (Just to prevent calling repeated times .Dispose() on JS side)

**Comment by lahma** [comment, marked as answer]
I think that would be a bit leaky abstraction, `delete` and and Dispose pattern don't map that well together. [The explicit resource management proposal](https://github.com/tc39/proposal-explicit-resource-management) would be something that would naturally map to .NET side.

Maybe you could have a helper function that wraps try-finally and Dispose following the actual semantics that .NET CLR/compiler does, something like:

```javascript
function usingDisposable(target, action) {
  try {
    action(target);
  }
  finally {
	target.dispose();
 }
}
```
